pkg/plugin/aggregation/aggregator.go

```diff
@@ -198,21 +204,47 @@ func (a *Aggregator) processResult(result *plugin.Result) error {
 	// Send an event that we got this result even if we get an error, so
 	// that Wait() doesn't hang forever on problems.
 	defer func() {
-		a.Results[result.ExpectedResultID()] = result
+		a.Results[result.Key()] = result
 		a.resultEvents <- result
 	}()
 
 	if err := a.handleResult(result); err != nil {
 		// Drop a breadcrumb so that we reconsider new results from this result.
-		a.FailedResults[result.ExpectedResultID()] = time.Now()
+		a.FailedResults[result.Key()] = time.Now()
 		return &httpError{
 			err:  fmt.Errorf("error handling result %v: %v", resultID, err),
 			code: http.StatusInternalServerError,
 		}
 	}
 
 	// Upon success, we no longer want to keep processing duplicate results.
-	delete(a.FailedResults, result.ExpectedResultID())
+	delete(a.FailedResults, result.Key())
+
+	return nil
+}
+
+// TODO
+func (a *Aggregator) processStatus(status plugin.StatusUpdate) error {
+	a.resultsMutex.Lock()
+	defer a.resultsMutex.Unlock()
+
+	// jschnake need to define how this expected stuff works. Either preload another
+	// expected status stuff or utilize the results work. but the results work expects
+	// the 'global' to be known due to a different URL path which we dont have.
+	// Seems like we'd have to say either the key (name+node) or it is (name+global)? Kind of wonky though
+	// and what if you get name/node and name/node2 but expected name+global? Or do we push that
+	// understanding down into the worker? Have a new env that says it is part of a job/ds?
```

Comment:
Old comment; can be removed. It now . will check about `expected` but the URL the plugins write to has the node/global in it so there is no ambiguity. I just followed the existing pattern used for results.